projects/ADuCM3029_demo_cn0503/src/platform_source/flash.c

```diff
@@ -112,7 +112,7 @@ static int32_t flash_write_page(struct flash_dev *dev, uint32_t flash_addr,
 		   FLASH_PAGE_SIZE_WORDS);
 
 	for(i = 0; i < array_size; i++)
-		temp_ptr[(i + (flash_addr & 0x7FF))] = array[i];
+		temp_ptr[(i + (flash_addr & 0x7FF) / 4)] = array[i];

```

Comment:
Curious, can you describe a bit what sort of bugs you encountered here?

Reply:
Flash writes were hanging for addresses that were not page-aligned. First, the while loop in flash_write was an infinite loop because temp (the amount by which the data index was being incremented) got set to 0. Then, it would hang or error on adi_fee_GetPageNumber in some cases. Not sure why that was happening, but being careful about whether flash_addr referred to bytes or words fixed it.